compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -595,16 +595,22 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             "create_substs_for_associated_item(span: {:?}, item_def_id: {:?}, item_segment: {:?}",
             span, item_def_id, item_segment
         );
-        self.create_substs_for_ast_path(
+        let (args, _) = self.create_substs_for_ast_path(
             span,
             item_def_id,
             parent_substs,
             item_segment,
             item_segment.args(),
             item_segment.infer_args,
             None,
-        )
-        .0
+        );
+
+        let assoc_bindings = self.create_assoc_bindings_for_generic_args(item_segment.args());
+        if let Some(b) = assoc_bindings.first() {
```

Comment:
I seems wasteful to convert all bindings, just to err on the first one later.  This could as well be implemented as
`if let Some(b) = item_segment.args().bindings.first()`
Probably better as a follow-up, as there are several instances of this pattern in the file.